ExampleMod/Common/GlobalTiles/AnimatedLoomGlobalTile.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.Xna.Framework.Graphics;
+using ReLogic.Content;
+using Terraria;
+using Terraria.GameContent;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Common.GlobalTiles
+{
+	//This class showcases how to change the vanilla loom texture and also add animation for our own texture
+	public class AnimatedLoomGlobalTile : GlobalTile
+	{
+		public const int FrameCount = 4;
+		public const int FrameSpeed = 16;
+
+		//We have to cache the textures we work with so we can reference them easier
+		private static Asset<Texture2D> originalLoom;
+
+		private static Asset<Texture2D> animatedLoom;
+
+		private static bool replacedTexture = false;
+
+		public override void SetupContent() {
+			//This hook is ran once on mod load, after all content has been loaded into the game, to be sure that everything exists
+			if (!Main.dedServ) {
+				//Do not try to do texture stuff on the server!
+				int type = TileID.Loom;
+
+				Main.instance.LoadTiles(type); //Loads the tile texture
+				originalLoom = TextureAssets.Tile[type]; //Cache the tile texture so we can restore it later on unload
+
+				animatedLoom = ModContent.GetTexture("ExampleMod/Common/GlobalTiles/AnimatedLoom"); //Fetch our own texture
+				TextureAssets.Tile[type] = animatedLoom; //Replace the vanilla texture with ours
+				replacedTexture = true; //Flag it so we can be sure we replaced it
```

Comment:
Seems a bit unneeded? Can't you just make it a Get and check if the texture matches the EM path? (If it is needed at all)

Reply:
sadly an ".rawimg" is appended to the assets Name property, and i don't know if it will stay like that implementation wise. I'd have to also concider the way the name is sanitized (don't know exactly off the top of my head) to then do a match with the path.